pkg/querier/querier_test.go

```diff
@@ -224,10 +310,95 @@ func TestQuerier_QueryableReturnsChunksOutsideQueriedRange(t *testing.T) {
 	}, m[0].Points)
 }
 
-func mockTSDB(t *testing.T, mint model.Time, samples int, step, chunkOffset time.Duration, samplesPerChunk int) (storage.Queryable, model.Time) {
+// TestBatchMergeChunks is a regression test to catch one particular case
+//
+//	when the Batch merger iterator was corrupting memory by not copying
+//	Batches by value because the Batch itself was not possible to copy
+//	by value.
+func TestBatchMergeChunks(t *testing.T) {
```

Comment:
that looks a bit strangely formatted - extra new line, tabs. Was this intentional?

Reply:
also - is this test testing `batch.mergeIterator`? Will the test fit better in that package instead?

Reply:
The bug was in the batch.mergeIterator, but it was complicated to trigger it directly. I used to have a test there as well, but we decided to keep only this one. Might catch something else later. Let's keep it here.